libraries/lib-utility/MemoryX.h

```diff
@@ -400,4 +401,40 @@ constexpr IntType SwapIntBytes(IntType value) noexcept
    return value;
 }
 
+//! Returns a rounded up integer result for numerator / denominator.
+/*!
+* RoundUpUnsafe(4, 2) == 2;
+* RoundUpUnsafe(3, 2) == 2;
+* RoundUpUnsafe(-3, 2) == -1;
+*
+* This function does not check if denominator is 0 of if there is an integer overflow. 
+*/
+template <typename LType, typename RType>
+auto RoundUpUnsafe(LType numerator, RType denominator) noexcept
+{
+   static_assert(std::is_integral_v<LType>);
+   static_assert(std::is_integral_v<RType>);
+
+   if constexpr (std::is_unsigned_v<LType> && std::is_unsigned_v<RType>)
+   {
+      return (numerator + denominator - 1) / denominator;
+   }
+   else
+   {
+      if (numerator > 0 && denominator > 0)
+      {
+         return (numerator + denominator - 1) / denominator;
+      }
+      else
+      {
+         const auto result = numerator / denominator;
+
+         if (result < 0 || result * denominator == numerator)
+            return result;
+         else
+            return result + 1;
+      }
+   }
+}
+
```

Comment:
I don’t like adding to MemoryX.h.  These might be in a new small header in lib math

Reply:
Sure